lib/cn/csched_sp3.c

```diff
@@ -891,12 +892,41 @@ sp3_unlink_all_nodes(struct sp3 *sp, struct cn_tree *tree)
 static_assert(NELEM(((struct sp3_node *)0)->spn_rbe) == RBT_MAX,
               "number of elements of spn_rbe[] is not RBT_MAX");
 
+/* "Scatter" is a measurement of the contiguity in virtual memory of a kvset's
+ * values relative to its keys.  For example, a kvset with (scatter == 1) means
+ * that for every key (n), the value for key (n+1) will immediately follow the
+ * value for key (n) in virtual memory.  The probability that the preceding is
+ * true decreases as the scatter increases.  Similarly, the probability that
+ * accessing a value will incur a TLB miss or a page fault is directionally
+ * proportional to scatter.
+ *
+ * Scatter is a direct consequence of k-compaction, where each k-compaction
+ * will typically amplify scatter by 4x or more.  Conversely, a kv-compaction
+ * completely eliminates scatter, returning the measurement to 1.
+ */
+static uint
+sp3_scatter(struct cn_tree_node *tn)
+{
+    struct kvset_list_entry *le;
+    uint scatter = 0;
+
+    list_for_each_entry(le, &tn->tn_kvset_list, le_link) {
+        uint vgroups = kvset_get_vgroups(le->le_kvset);
```

Comment:
add const :)

Reply:
Will do if I need to make another substantive edit :-)